sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultMetaData.java

```diff
@@ -155,6 +171,29 @@ private String getTableName(final int columnIndex) throws SQLException {
         Optional<TableRule> tableRule = shardingRule.findTableRuleByActualTable(actualTableName);
         return tableRule.isPresent() ? tableRule.get().getLogicTable() : actualTableName;
     }
+
+    private String getTableName(final int columnIndex, final SQLStatementContext sqlStatementContext) throws SQLException {
+        final Collection<String> tableNames = sqlStatementContext.getTablesContext().getTableNames();
+        for (String each : tableNames) {
+            if (isCurrentTable(columnIndex, each)) {
+                return each;
+            }
+        }
+        return "";
+    }
+
+    private boolean isCurrentTable(final int columnIndex, final String logicTableName) throws SQLException {
+        Collection<String> actualColumns = encryptRule.findEncryptTable(logicTableName)
+                .transform(new Function<EncryptTable, Collection<String>>() {
+                    @Override
+                    public Collection<String> apply(final EncryptTable encryptTable) {
+                        return encryptTable.getCipherColumns();
+                    }
+                })
+                .or(Collections.<String>emptyList());
+        String actualColumn = resultSetMetaData.getColumnName(columnIndex);
```

Comment:
Hi, to get actualColumn, you use `resultSetMetaData.getColumnName(columnIndex)` which is called in `getLogicColumn(final String tableName, final int columnIndex)` as well.
**Notice** The purpose of getShardingEncryptor(final int columnIndex) is to get Optional<ShardingEncryptor>, so i think you can merge `isCurrentTable()` and `getLogicColumn(final String tableName togather to get the final result, i.e, Optional<ShardingEncryptor>.

Reply:
Good idea, i will modify it like this!